### PR TITLE
Inventory, Debug Label Background & Screenshots folder

### PR DESCRIPTION
**Pull request by Arne1303**
Hey there, 
I added an Inventory with Drag & Drop functionality. It allows you to drag Blocks into your Hotbar, there are also 3 new Configs associated with the inventory,

- HOTBAR_SIZE (how many items are shown in your Hotbar)
- INVENTORY_MARGIN (Spacing between items)
- HOTBAR_SIZE (PREVIEW_SIZE size of each inventory slot)

Have fun experimenting with this!

### CODE REVIEW

File: game/inventory.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+        self.hotbar_background = VertexRectangle(0, 0, 100, 100, (90, 101, 117, 125))
+        self.selectionIndicator = VertexRectangle(0, 0, PREVIEW_SIZE, PREVIEW_SIZE, (0, 0, 0, 80))
+
+        #Last known possition of mouse and hovered item
+        self.mouse_x = 0
+        self.mouse_y = 0
+        self.drag_x = 0
+        self.drag_y = 0
+        self.hovered_item = None
+        self.is_draggign = False
```
Comment (benmoran56) [inline game/inventory.py]: Small typo: is_dragging

**Comment by benmoran56** [comment]
Thanks a lot for the pull request! It looks good overall.

I have a few small requests if you don't mind. 
If possible, could you rename the camel case variables to fit with the typical Python coding style? For example, instead of:  `selectionIndicator`, `newBlock`, `debugBatch`, etc., something like `selection_indicator`, `new_block`/`newblock` or something along those lines. 

Also, in the last week, pyglet got a new `pyglet.shapes` module. This includes Rectangles with the ability of adding them to the main Batch. The explicit setting of GL_BLEND before the main app loop is also not required, since the Shapes internal Group handles that for us. 

Let me know if you have any questions about anything, or if you don't have time to make the changes I can take care of it myself after merging. 

Thanks again!

**Comment by Arne1303** [comment]
yes, I can do both! Do you happen to know if Pyglet supports 9 tiled sprites? 
Would be great for the background in the inventory, 
if not I could definitely write something for it.

**Comment by Arne1303** [comment]
I removed the Vertex Rectangle and tried to sort everything in the inventory somewhat in batches, 
but the performance with the inventory seems to be extremely bad ... do you have any ideas on how to improve the performance?

**Comment by benmoran56** [comment]
I took a deep dive into it, and it looks like there are lots of things being updated every frame. Many things are being constantly recalculated, even if the inventory is not open. 

If the game is paused while the inventory is open, it seems like it might make more sense to reorganize it as a Scene that is switched to/from, rather than being integrated into the GameScene. What do you think about that?

**Comment by Arne1303** [comment]
I have improved performance quite a bit now, and I think the impact is no longer that noticeable. I would say we keep it like this and move it to another Screen when there are any more problems with it.

One more question, how do you turn off the Reset of the Texture scaling after using the pyglet text module? U mention it in this conversation  #42

**Comment by Arne1303** [comment]
@benmoran56 how are we gonna continue with this pr? For me, it does run fine now

**Pull request closed without merging**